### PR TITLE
Update jandex version.

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

File: environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileInfo.java [hunk 1]
```diff
@@ -230,8 +228,8 @@ private boolean isAssignableTo(DotName name, Class<?> to) {
             return true;
         }
 
-        if (fromClassInfo.interfaces() != null) {
```
Comment (tremes) [inline environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileInfo.java]: this method was deprecated

File: environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileInfo.java [hunk 2]
```diff
@@ -106,9 +106,7 @@ public boolean isVetoed() {
 
     @Override
     public boolean isTopLevelClass() {
-        // TODO This is not portable per the JSL
-        // TODO Modify jandex to contain isTopLevelClass attribute
-        return !classInfo.name().local().contains("$");
+        return !classInfo.name().isInner();
```
Comment (mkouba) [inline environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileInfo.java]: I think we should rather use `ClassInfo.nestingType()` here.

**Comment by mkouba** [comment]
We should also uncomment the relevant section in `org.jboss.weld.util.Beans.isTypeManagedBeanOrDecoratorOrInterceptor(ClassFileInfo)`.

**Comment by mkouba** [comment]
Hm, that's odd. Uncommenting the check caused all the static nested classes to be ignored on WildFly.

**Comment by mkouba** [comment]
https://issues.jboss.org/browse/JANDEX-37?focusedCommentId=13205518&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13205518

**Comment by tremes** [comment]
Closed in favor of https://github.com/weld/core/pull/1440

**Pull request closed without merging**